src/Client/AzureManaged/DurableTaskSchedulerClientOptions.cs

```diff
@@ -0,0 +1,158 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System.ComponentModel.DataAnnotations;
+using Azure.Core;
+using Azure.Identity;
+using Grpc.Core;
+using Grpc.Net.Client;
+
+namespace Microsoft.DurableTask;
+
+/// <summary>
+/// Options for configuring the Durable Task Scheduler.
+/// </summary>
+public class DurableTaskSchedulerClientOptions
+{
+    /// <summary>
+    /// Gets or sets the endpoint address of the Durable Task Scheduler resource.
+    /// Expected to be in the format "https://{scheduler-name}.{region}.durabletask.io".
+    /// </summary>
+    [Required(ErrorMessage = "Endpoint address is required")]
+    public string EndpointAddress { get; set; } = string.Empty;
+
+    /// <summary>
+    /// Gets or sets the name of the task hub resource associated with the Durable Task Scheduler resource.
+    /// </summary>
+    [Required(ErrorMessage = "Task hub name is required")]
+    public string TaskHubName { get; set; } = string.Empty;
+
+    /// <summary>
+    /// Gets or sets the credential used to authenticate with the Durable Task Scheduler task hub resource.
+    /// </summary>
+    public TokenCredential? Credential { get; set; }
+
+    /// <summary>
+    /// Gets or sets the resource ID of the Durable Task Scheduler resource.
+    /// The default value is https://durabletask.io.
+    /// </summary>
+    public string ResourceId { get; set; } = "https://durabletask.io";
+
+    /// <summary>
+    /// Gets or sets a value indicating whether to allow insecure channel credentials.
+    /// This should only be set to true in local development/testing scenarios.
+    /// </summary>
+    public bool AllowInsecureCredentials { get; set; }
+
+    /// <summary>
+    /// Creates a new instance of <see cref="DurableTaskSchedulerClientOptions"/> from a connection string.
+    /// </summary>
+    /// <param name="connectionString">The connection string to parse.</param>
+    /// <returns>A new instance of <see cref="DurableTaskSchedulerClientOptions"/>.</returns>
+    public static DurableTaskSchedulerClientOptions FromConnectionString(string connectionString)
+    {
+        return FromConnectionString(new DurableTaskSchedulerConnectionString(connectionString));
+    }
+
+    /// <summary>
+    /// Creates a gRPC channel for communicating with the Durable Task Scheduler service.
+    /// </summary>
+    /// <returns>A configured <see cref="GrpcChannel"/> instance that can be used to make gRPC calls.</returns>
+    public GrpcChannel CreateChannel()
```

Comment:
This shouldn't be public.

```suggestion
    internal GrpcChannel CreateChannel()
```

If you need to access it for testing you can add InternalsVisibleTo like so:

https://github.com/microsoft/durabletask-dotnet/blob/main/src/Abstractions/Abstractions.csproj#L22